Lib/test/test_capi/test_opt.py

```diff
@@ -2105,17 +2105,76 @@ def testfunc(n):
         self.assertNotIn("_TO_BOOL_BOOL", uops)
         self.assertIn("_GUARD_IS_TRUE_POP", uops)
 
-    def test_call_isinstance_tuple_of_classes(self):
+    def test_call_isinstance_tuple_of_classes_is_true(self):
         def testfunc(n):
             x = 0
             for _ in range(n):
-                # A tuple of classes is currently not optimized,
-                # so this is only narrowed to bool:
                 y = isinstance(42, (int, str))
                 if y:
                     x += 1
             return x
 
+        res, ex = self._run_with_optimizer(testfunc, TIER2_THRESHOLD)
+        self.assertEqual(res, TIER2_THRESHOLD)
+        self.assertIsNotNone(ex)
+        uops = get_opnames(ex)
+        self.assertNotIn("_CALL_ISINSTANCE", uops)
+        self.assertNotIn("_TO_BOOL_BOOL", uops)
+        self.assertNotIn("_GUARD_IS_TRUE_POP", uops)
+        self.assertIn("_BUILD_TUPLE", uops)
+        self.assertIn("_POP_CALL_TWO_LOAD_CONST_INLINE_BORROW", uops)
```

Comment:
`_BUILD_TUPLE` is preventing us from optimizing out `_POP_CALL_TWO_LOAD_CONST_INLINE_BORROW`.
The bytecode is basically:

```c
LOAD_CONST
LOAD_CONST
_BUILD_TUPLE
_POP_CALL_TWO_LOAD_CONST_INLINE_BORROW
```

To optimize this, we'd need some special handling for `_BUILD_TUPLE` in `remove_unneeded_uops`.

Reply:
Got it, might be worth looking into next if you're up for it! Could be tricky, though.